Feat: add exchange key update API

SeijaHuang opened the pull request:
Added exchange key update API
-	Added update exchange key input validation and its corresponding unit test
-	Added update exchange key resolver
-	Added service method and its corresponding unit test
-	Move the exchangeKey validation pipe to the resolver level so it can be applied to both the create and update APIs Resolve CP-11

Postman test:
1. Id is empty:
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/0bc3c0b0-4881-4b98-8001-d03bf7f3fe03">

2. Id is invalid:
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/58637bcb-9faf-47f7-9ade-cf1c81437826">

3. The exchange key is not found
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/6bd0e286-f7a6-4db8-b2a8-1eabd4a90eb7">

4. New exchange key is invalid
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/5ba4d99e-622b-4d3f-bb90-7c7a351be922">

5. Success
<img width="1411" alt="image" src="https://github.com/user-attachments/assets/e6d61e76-3b37-4fbf-8bc0-9f349225b85d">

Test result:
<img width="686" alt="image" src="https://github.com/user-attachments/assets/d34f92ee-1d00-4836-a2e3-22f1ec82373d">

Chris-Wang reviewed (changes requested):
Resolve conflict and fix failed pipeline

SeijaHuang committed:
Filter changed files
Feat: add exchange key update API
Added exchange key update API
- Added update exchange key input validation and its corresponding unit test
- Added update exchange key resolver
- Added service method and its corresponding unit test
- Move the exchangeKey validation pipe to the resolver level so it can be applied to both the create and update APIs
Resolve CP-11

make changes to testing file

Chris-Wang reviewed (approved):
LGTM
